Merge_Sort_C/main.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+
+void PrintArray(int array[], int sizeOfArray)
+{
+    for(int i = 0; i < sizeOfArray; i++)
+    {
+        printf("%d  ", array[i]);
+    }
+    printf("\n");
+}
+
+void MergeArrays(int array[], int leftHalf[], int rightHalf[], int left, int right)
+{
+    int i = 0;
+    int j = 0;
+    int k = 0;
+
+    while (i < left && j < right) {
+        if(leftHalf[i] <= rightHalf[j])
+        {
+            array[k++] = leftHalf[i++];
+        }
+        else
+        {
+            array[k++] = rightHalf[j++];
+        }
+    }
+    while (i < left)
+    {
+        array[k++] = leftHalf[i++];
+    }
+    while (j < right)
+    {
+        array[k++] = rightHalf[j++];
+    }
+}
+
+void SortArray(int array[], int sizeOfArray)
+{
+    if(sizeOfArray < 2) return;
+
+    int beforeMiddle = sizeOfArray / 2;
```

Comment:
Variable name isn't obvious. I would suggest to call them smth like `leftArraySize` and `rightArraySize`